streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/TimeWindowedKStream.scala

```diff
@@ -58,11 +58,8 @@ class TimeWindowedKStream[K, V](val inner: TimeWindowedKStreamJ[K, V]) {
    * represent the latest (rolling) count (i.e., number of records) for each key
    * @see `org.apache.kafka.streams.kstream.TimeWindowedKStream#count`
    */
-  def count()(implicit materialized: Materialized[K, Long, ByteArrayWindowStore]): KTable[Windowed[K], Long] = {
-    val c: KTable[Windowed[K], java.lang.Long] =
-      inner.count(materialized.asInstanceOf[Materialized[K, java.lang.Long, ByteArrayWindowStore]])
-    c.mapValues[Long](Long2long _)
-  }
+  def count()(implicit materialized: Materialized[K, Long, ByteArrayWindowStore]): KTable[Windowed[K], Long] =
```

Comment:
Also worth noting that the same problem afflicts `SessionWindowedKStream.scala`. Can you add coverage for that class in your test as well?

Reply:
Good catch. Updated, and added a unit test that is effectively a copy of

https://github.com/apache/kafka/blob/04e206154ac614b7d4d34a7a1b6ba2c882f607b9/streams/src/test/java/org/apache/kafka/streams/kstream/internals/SuppressScenarioTest.java#L463